### PR TITLE
Expands the info about legacy database support

### PR DESCRIPTION
**Pull request by mereghost**
Adding some info on how to deal with legacy databases, the caveats of mapping and working with just a subset of the schema.

I'm open to suggestions.

### CODE REVIEW

File: source/guides/1.1/repositories/overview.md [hunk 1]
```diff
@@ -218,6 +217,72 @@ operator = OperatorRepository.new.create(name: "Jane")
...
+    attribute :s_name, Hanami::Model::Sql::Types::String.meta(alias: :name)
+  end
+  
+  mapping do
+    attribute :id,   from: :operator_id
+    attribute :name, from: :s_name
+  end
+  
+  def by_name(name)
+    t_operator.where(name: name)
```
Comment (davydovanton) [inline source/guides/1.1/repositories/overview.md]: can we use `#root` here?

File: source/guides/1.1/repositories/overview.md [hunk 2]
```diff
@@ -218,6 +217,72 @@ operator = OperatorRepository.new.create(name: "Jane")
...
+```
+
+If you want to use the more semantic names (or with just a subset of attributes) you'll need to define a schema and a mapping:
+
+```ruby
+class OperatorRepository < Hanami::Repository
+  self.relation = :t_operator
+  
+  schema do
+    attribute :operator_id, Hanami::Model::Sql::Types::Int.meta(primary_key: true, alias: :id)
```
Comment (davydovanton) [inline source/guides/1.1/repositories/overview.md]: just idea, but maybe we can use more friendly types here?

**Comment by AlfonsoUceda** [comment]
@mereghost can you ask to @davydovanton I think after that we can merge it.

**Comment by mereghost** [comment]
@AlfonsoUceda Sure. I was away-ish the last few days. I'll address his points and update this.

**Comment by cllns** [comment]
Cleaned up some of the language, but I'm not comfortable enough with this material to address @davydovanton's points :)

File: source/guides/1.1/repositories/overview.md (hunk 1 above)
Comment (mereghost) [reply source/guides/1.1/repositories/overview.md]: sure.

File: source/guides/1.1/repositories/overview.md (hunk 2 above)
Comment (mereghost) [reply source/guides/1.1/repositories/overview.md]: While we probably could use `Dry::Types` not sure if it wouldn't lead to confusion as the end user might think something along the likes of "oh I need yet another lib to do this".

**Comment by mereghost** [comment]
@cllns Thanks for the help. =)

**Pull request merged**